src/test/rustdoc-ui/intra-doc/ambiguity.stderr

```diff
@@ -83,19 +83,19 @@ LL | /// Ambiguous [const@type_and_value].
    |                ^^^^^^^^^^^^^^^^^^^^
 
 error: `foo::bar` is both an enum and a function
-  --> $DIR/ambiguity.rs:35:42
+  --> $DIR/ambiguity.rs:35:43
    |
 LL | /// Ambiguous non-implied shortcut link [`foo::bar`].
-   |                                          ^^^^^^^^^^ ambiguous link
+   |                                           ^^^^^^^^ ambiguous link
    |
 help: to link to the enum, prefix with `enum@`
    |
 LL | /// Ambiguous non-implied shortcut link [`enum@foo::bar`].
-   |                                          ^^^^^^^^^^^^^^^
+   |                                           ^^^^^^^^^^^^^
 help: to link to the function, add parentheses
    |
 LL | /// Ambiguous non-implied shortcut link [`foo::bar()`].
-   |                                          ^^^^^^^^^^^^
+   |                                           ^^^^^^^^^^
```

Comment:
Given that these are being shown as verbose already (because there's more than one suggestion), it'd be nice to make the span smaller, only covering the `enum@` and `()`.

Reply:
This is an interesting take. Although, not sure it would be an improvement in these cases because they ask to add things, not remove them.

Reply:
This would look like this:

```
error: `foo::bar` is both an enum and a function
  --> $DIR/ambiguity.rs:35:43
   |
LL | /// Ambiguous non-implied shortcut link [`foo::bar`].
   |                                           ^^^^^^^^ ambiguous link
   |
help: to link to the enum, prefix with `enum@`
   |
LL | /// Ambiguous non-implied shortcut link [`enum@foo::bar`].
   |                                           ^^^^^
help: to link to the function, add parentheses
   |
LL | /// Ambiguous non-implied shortcut link [`foo::bar()`].
   |                                                   ^^
```

which to my eyes looks easier to understand at a glance.